src/tablestore.rs

```diff
@@ -446,4 +515,130 @@ mod tests {
         let result = ts.write_sst(&wal_id, table2).await;
         assert!(matches!(result, Err(error::SlateDBError::Fenced)));
     }
+
+    #[tokio::test]
+    async fn test_tablestore_sst_and_partial_cache_hits() {
+        // Setup
+        let os = Arc::new(InMemory::new());
+        let format = SsTableFormat::new(32, 1, None);
+        let block_cache = Arc::new(BlockCache::builder().max_capacity(1024).build());
+        let ts = Arc::new(TableStore::new(
+            os.clone(),
+            format,
+            Path::from("/root"),
+            Some(block_cache.clone()),
+        ));
+
+        // Create and write SST
+        let id = SsTableId::Compacted(Ulid::new());
+        let mut writer = ts.table_writer(id);
+        for i in 0..20 {
+            let key = [i as u8; 16];
+            let value = [(i + 1) as u8; 16];
+            writer.add(&key, Some(&value)).await.unwrap();
+        }
+        let handle = writer.close().await.unwrap();
+
+        // Read the index
+        let index = Arc::new(ts.read_index(&handle).await.unwrap());
+
+        // Test 1: SST hit
+        let blocks = ts
+            .read_blocks_using_index(&handle, index.clone(), 0..20, true)
+            .await
+            .unwrap();
+
+        let mut expected_data = Vec::with_capacity(20);
+        for i in 0..20 {
```

Comment:
Nit: might be easier to assemble expected_data as part of the previous for loop.